### ISSUE TITLE
(Inner/outer) Horizontal alignment as core option

### ISSUE THREAD

**Issue by strarsis** [WordPress/gutenberg#20141]
**Is your feature request related to a problem? Please describe.**
Currently, it seems to be very hard to horizontally align e.g. a `core/group` block to the left, center or right. Using CSS it is very easy, for outer horizontal alignment (block relative to parent), it is sufficient to use some CSS like `(display: block; )margin-left: 0; margin-right: auto;`. But there is no UI in Gutenberg for this.
Inner alignment seems to be a more niche case because then the outer block element must have different constraints (e.g. `max-width`) than its inner wrapper or content. In one theme I had to use this approach and make the outer element of  `wide` `group` blocks as wide as possible to extend background fx, but still constraint their inner wrapper element to a reasonable `max-width`.

**Describe the solution you'd like**
UI controls for outer (and if possible or optional inner) horizontal alignment for blocks.

**Describe alternatives you've considered**
The alternative would be manually applying CSS classes to blocks.

**Comment by draganescu** [comment]
Hello @strarsis, is this a request that the `Group` block have the same alignment controls that for example the `Image` block has:

<img width="686" alt="Screenshot 2020-02-11 at 15 45 58" src="https://user-images.githubusercontent.com/107534/74241916-9fc60d00-4ce5-11ea-9e9d-1d8c8642b48c.png">

**Comment by strarsis** [comment]
@draganescu: Yes, this would be great! As a class is applied for the alignment, the theme can either align the inner/contents of the group block or the outside alignment of group block (by making it "push"), depending on how the theme styles the `wide`/`full` group blocks.
For example, a group block may extend its background (outer wrapper element) completely without constrains and provide a visual background, while its inner wrapper element with content is aligned to the left/center/right.

**Comment by rickybanister** [comment]
Related: #24473

**Comment by mtias** [comment]
This should be covered by this:

<img width="350" height="123" alt="Image" src="https://github.com/user-attachments/assets/1b7f4b50-0ae5-47ce-ad2d-18ce5e09e9c9" />
